corehq/apps/app_manager/suite_xml.py

```diff
@@ -2348,9 +2348,11 @@ def menus(self):
                 def get_commands():
                     for form in module.get_forms():
                         command = Command(id=id_strings.form_command(form))
-                        if module.all_forms_require_a_case() and \
-                                not module.put_in_root and \
-                                getattr(form, 'form_filter', None):
+                        if (
+                            (is_usercase_in_use(self.app.domain) or module.all_forms_require_a_case()) and
+                            not module.put_in_root and
+                            getattr(form, 'form_filter', None)
+                        ):
```

Comment:
I think the top level criteria here is:
```python
not module.put_in_root and getattr(form, 'form_filter', None)
```
Once that has been determined the only other criteria that matters is `module.all_forms_require_a_case()` which should determine whether the user can use `#case` or `./` in the expression